src/python/gudhi/tensorflow/_layer.py

```diff
@@ -0,0 +1,16 @@
+import tensorflow
+# The sklearn parse_version function is taken from 'packaging' and this dependency is not mandatory
+from sklearn.utils.fixes import parse_version
+
+class TensorflowKerasLayer(tensorflow.keras.layers.Layer):
+    """
+    TensorFlow Keras layer
+    """
+    def __init__(self, **kwargs):
+        """
+        Constructor for the TensorflowKerasLayer class
+        """
+        # On tensorflow < 2.16.0 set dynamic argument to True
+        if parse_version(tensorflow.__version__) < parse_version('2.16.0'):
```

Comment:
#1048 says that 2.15.1 rejects `dynamic=True`, I am confused...

Reply:
I tried again with master branch of gudhi and python 3.11. The problem seems to occur on tensorflow >= 2.16.
After reading the [tensorflow 2.16.0 release note](https://github.com/tensorflow/tensorflow/releases/tag/v2.16.0-rc0), this change seems to come from `keras.layers.experimental.DynamicEmbedding` change.

Sorry about that, I was pretty sure it happens on tensorflow 2.15.1, but after several tests I can guarantee it happens on tensorflow 2.16.1 (no pip package for tensorflow 2.15.0)